aexpect/ops_linux.py

```diff
@@ -237,6 +237,30 @@ def ls(session, path, quote_path=True, flags="-1UNq"):  # pylint: disable=C0103
     return output.splitlines()
 
 
+def glob(session, glob_pattern):
+    """
+    Find files matching given pattern through a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str glob_pattern: pattern of filenames where `*`, `?` and such match
+                             any string or not-empty-string and so on.
+    :returns: file names matching pattern, including path (like glob.glob)
+    :rtype: [str]
+
+    The current remote implementation has the limitation that it would hide
+    any `ls` exit status 2 that is a serious error not related to the glob
+    expansion. Alternative implementations are either shell specific or use
+    `echo` and thus cannot handle spaces in paths well.
+    """
+    cmd = f'ls {glob_pattern}'
+    status, output = session.cmd_status_output(cmd)
+    if status == 2:
+        return []
+    status, output = _process_status_output(cmd, status, output)
+    return output.splitlines()
```

Comment:
In order to be able to split by lines it's necessary to enforce `-1` argument:

```
>>> aexpect.ops_linux.ls(a, "*", False, '-1UNq')
['a', 'a b']
>>> aexpect.ops_linux.ls(a, "*", False, '')
["a  'a b'"]
```

Also note that the `ls` approach can result in surprising output when in use with dirs:

```
>>> aexpect.ops_linux.ls(a, "/tmp/b", False, '-1')
['a', 'b']
>>> aexpect.ops_linux.ls(a, "/tmp/b/*", False, '-1')
['/tmp/b/a:', '', '/tmp/b/b:']
```

I don't want to draw the line here, it depends on which features you, as the primary user, need. I'm fine with any approach as long as it gets properly documented and is somehow systematic. So I'd suggest the use of glob and arrays, or ls with the usual arguments (perhaps just call the `ops_linux.ls` and catch for status 2?)

Reply:
All good points and integration tests here are still running hence I didn't notice this yet. How about using something that will simplify all of these residues of user formatted output like `find`?
```
[pevogam@drogon androids]$ find c*
c
c b
[pevogam@drogon androids]$ ls c*
'c b'

c:
```
I guess it depends what the user would expect to get under glob but I would reckon receiving a guaranteed list of paths seems much more reasonable expectation than dir-ls-like behavior.

Reply:
(the `-1` was indeed important but my testing didn't catch it because of some default `ls` settings on some linux distros)

Reply:
I though about find for a while but decided not to pursuit it further, but it can also work. Only I'd suggest using `-maxdepth 0`. With that it works well for all of the variants I tried.

Reply:
find /tmp/*/*a* -maxdepth 0

Reply:
find /tmp/b/* -maxdepth 0

Reply:
Thanks and I apologize for the micromanagement here - pushed and CI passing, will need more time to confirm the integration tests we have but I think we are both feeling confident about the final solution now.

Reply:
No problem, that is what reviews are for.